### PR TITLE
Add in course type and run types to course options call

### PR DESCRIPTION
**Pull request by Dillon-Dumesnil**
DISCO-1386

### CODE REVIEW

File: course_discovery/apps/api/serializers.py [hunk 1]
```diff
@@ -165,9 +165,28 @@ def determine_metadata(self, request, view):
         self.view = view  # pylint: disable=attribute-defined-outside-init
         return super().determine_metadata(request, view)
 
+    def create_type_options(self, info):
+        info['type_options'] = [{
+            'uuid': course_type.uuid,
+            'name': course_type.name,
+            'entitlement_types': [entitlement_type.slug for entitlement_type in course_type.entitlement_types.all()],
```
Comment (mikix) [inline course_discovery/apps/api/serializers.py]: Just the slug, not a name for them either?

File: course_discovery/apps/api/v1/views/courses.py [hunk 2]
```diff
@@ -68,6 +68,7 @@ class CourseViewSet(CompressedCacheResponseMixin, viewsets.ModelViewSet):
     serializer_class = serializers.CourseWithProgramsSerializer
     metadata_class = MetadataWithRelatedChoices
     metadata_related_choices_whitelist = ('mode', 'level_type', 'subjects',)
+    metadata_nested_info_whitelist = ('type',)
```
Comment (mikix) [inline course_discovery/apps/api/v1/views/courses.py]: This code is so custom, I might say don't even attempt to look like you're a generic thing. Like, maybe `metadata_course_type_field = 'type'` or `metadata_show_course_type_info = True` or something dumb.

**Review by mikix — APPROVED** [review]

File: course_discovery/apps/api/v1/views/courses.py (hunk 2 above)
Comment (Dillon-Dumesnil) [reply course_discovery/apps/api/v1/views/courses.py]: I was debating it as well. I don't disagree. I'll update.

File: course_discovery/apps/api/serializers.py (hunk 1 above)
Comment (Dillon-Dumesnil) [reply course_discovery/apps/api/serializers.py]: Yeah, we don't actually surface this information. It is used in Frontend Applciation Publisher for
```
if (type.entitlement_types.length && type.entitlement_types[0] !== AUDIT_TRACK.key) {
```

File: course_discovery/apps/api/v1/views/courses.py (hunk 2 above)
Comment (mikix) [reply course_discovery/apps/api/v1/views/courses.py]: Or you know, use a subclass of MetadataWithRelatedChoices that added the logic. That's a reasonably clean way to isolate this code

Comment (Dillon-Dumesnil) [reply course_discovery/apps/api/v1/views/courses.py]: I like that more. One sec

File: course_discovery/apps/api/serializers.py [hunk 3]
```diff
@@ -165,6 +165,7 @@ def determine_metadata(self, request, view):
         self.view = view  # pylint: disable=attribute-defined-outside-init
         return super().determine_metadata(request, view)
 
+
```
Comment (mikix) [inline course_discovery/apps/api/serializers.py]: Might be accidental, but who cares.

**Review by mikix — APPROVED** [review]

**Pull request merged**